sdk/src/account_context.rs

```diff
@@ -0,0 +1,258 @@
+//! Successors of invoke_context::StackFrame, KeyedAccount and AccountInfo
+
+use crate::{
+    account::{AccountSharedData, ReadableAccount, WritableAccount},
+    instruction::InstructionError,
+    pubkey::Pubkey,
+};
+use std::cell::{RefCell, RefMut};
+
+/// Loaded transaction shared between runtime and programs.
+///
+/// This context is valid for the entire duration of a transaction being processed.
+pub struct TransactionContext {
+    account_keys: Vec<Pubkey>,
+    accounts: Vec<RefCell<AccountSharedData>>,
+    invoke_stack_max_height: usize,
+    invoke_stack: Vec<InstructionContext>,
+    return_data: (Pubkey, Vec<u8>),
+}
+
+impl TransactionContext {
+    pub fn new(
+        keys_and_accounts: Vec<(Pubkey, AccountSharedData)>,
```

Comment:
Why take a `Vec`?

Reply:
Otherwise there would be a slice reference in `TransactionContext` introducing a lifetime, both is not going to fly in ABIv2 because `TransactionContext` needs to be serializable, thus self contained.